lib/model/classycat_schema_lookup.py

```diff
@@ -74,4 +75,27 @@ def process(self, message: Message) -> ClassyCatSchemaResponse:
             return result
         except Exception as e:
             logger.error(f"Error looking up schema name {schema_name}: {e}")
-            raise PrestoBaseException(f"Error looking up schema name {schema_name}", 500) from e
\ No newline at end of file
+            raise PrestoBaseException(f"Error looking up schema name {schema_name}", 500) from e
+
+
+    @classmethod
+    def validate_input(cls, data: Dict) -> None:
+        """
+        Validate input data. Must be implemented by all child "Model" classes.
+        """
+        if "schema_name" not in data["parameters"] or data["parameters"]["schema_name"] == "":
+            raise PrestoBaseException("schema_name is required as input to schema look up", 422)
+
+    @classmethod
+    def parse_input_message(cls, data: Dict) -> Any:
+        """
+        Parse input into appropriate response instances.
+        """
+        event_type = data['parameters']['event_type']
+        result_data = data.get('result', {})
+
+        if event_type == 'schema_lookup':
+            return ClassyCatSchemaResponse(**result_data)
```

Comment:
seems like this needs to check that  either the  schema_id or schema_name are not empty?  It looks like ClassyCatSchemaResponse considers schema_id as optional?

Reply:
the class specific validator checks that those fields exist, look at `classycat_schema_lookup.py`

however this function as implemented before only searches by name, not both name and id. maybe we can file a ticket for that if it's necessary?

Reply:
I thought there was already a function for lookup by schema name and a separate one for id? (I  just wasn't sure which this was)